storage/repair.go

```diff
@@ -372,16 +382,31 @@ func (r *dbRepairer) Report() {
 	}
 }
 
-func (r *dbRepairer) repairWithTimeRange(tr xtime.Range) error {
-	multiErr := xerrors.NewMultiError()
-	namespaces := r.database.getOwnedNamespaces()
-	for _, n := range namespaces {
-		if err := n.Repair(r.shardRepairer, tr); err != nil {
-			detailedErr := fmt.Errorf("namespace %s failed to repair time range %v: %v", n.ID().String(), tr, err)
-			multiErr = multiErr.Add(detailedErr)
+func (r *dbRepairer) repairNamespaceWithTimeRange(n databaseNamespace, tr xtime.Range) error {
+	var (
+		rtopts    = n.Options().RetentionOptions()
+		blockSize = rtopts.BlockSize()
+		err       error
+	)
+
+	// repair the namespace
+	if err = n.Repair(r.shardRepairer, tr); err != nil {
+		err = fmt.Errorf("namespace %s failed to repair time range %v: %v", n.ID().String(), tr, err)
+	}
+
+	// update repairer state
+	for t := tr.Start; t.Before(tr.End); t = t.Add(blockSize) {
+		repairState, _ := r.repairStatesByNs.repairStates(n.ID(), t)
+		if err == nil {
+			repairState.Status = repairSuccess
+		} else {
+			repairState.Status = repairFailed
+			repairState.NumFailures++
 		}
+		r.repairStatesByNs.setRepairState(n.ID(), t, repairState)
```

Comment:
Haven't quite grokked the locking for the `dbRepairer` but thought should check in, does this need to be locked when get/set the repair states?  Doesn't seem like callers of this method necessarily lock.

Reply:
Yep we don't lock because the databaseRepairer [guarantees](https://github.com/m3db/m3db/blob/c1babe8c72d31c7b8a599c87ec1b6f18e88cc380/storage/repair.go#L358-L364) at most 1 repair is executing

Reply:
Does that mean we can remove all the other locks?  Perhaps we can document at the struct level which fields are protected by mutex and which ones protected by the run lock?

Reply:
Yep can move the lock to only lock a single field, will make the change.

Reply:
done